### DISCUSSION TITLE
StatusCode and StatusMessage of AdmissionResult

### DISCUSSION THREAD

**Discussion by anekdoti** [dotnet/dotnet-operator-sdk#287] [Q&A] [answered]
Why are the StatusCode and the StatusMessage of AdmissionResult (and therefore also e.g. ValidationResult) internal and there is no getter whatsoever for them, although they can be set by the user? It makes it very hard to write unit tests e.g. for a ValidationWebhook.

**Comment by buehler** [comment, marked as answer]
Uhm... to be honest, there is no real reason. I just thought that adding things to the public is not a breaking change while - for some reason - removing things is a breaking change. Therefore I just exposed everything that is absolutely necessary.

Do you think it would make things easier if those fields were public?

**Reply by anekdoti** [reply]
It would make it easier to write proper unit tests for the handler methods that check for the proper ValidationResult.

**Reply by buehler** [reply]
That's a good reason.
